src/main/java/tr/ogu/edu/school/schoolapp/service/InstallmentService.java

```diff
@@ -7,17 +7,22 @@
 import jakarta.transaction.Transactional;
 import lombok.RequiredArgsConstructor;
 import tr.ogu.edu.school.schoolapp.model.Installment;
+import tr.ogu.edu.school.schoolapp.model.User;
 import tr.ogu.edu.school.schoolapp.repository.InstallmentRepository;
 
 @Service
 @RequiredArgsConstructor
 public class InstallmentService {
 
 	private final InstallmentRepository installmentRepository;
+	private final AuthenticationService authenticationService;
 
-	public List<Installment> getInstallmentsByUserId(Long userId) {
-		// FIXME: Bu metod, oturum açan kullanıcının gerçek kimliği ile güncellenecek.
-		return installmentRepository.findInstallmentsByUserId(userId);
+	public List<Installment> getInstallmentsByUserId() {
+		User authenticatedUser = authenticationService.getAuthenticatedUser();
+		if (authenticatedUser == null) {
+			throw new IllegalStateException("Oturum açan kullanıcı bulunamadı.");
+		}
```

Comment:
Controller endpoint zaten filtrelendiği için bu şekilde null gelmesi söz konusu olmamalı. Bu sebeple bu kontrole gerek duymayız.